craft_application/errors.py

```diff
@@ -120,6 +120,26 @@ def __init__(self, platform: str, all_platforms: Sequence[str]) -> None:
         super().__init__(message=message, details=details)
 
 
+class EmptyBuildPlanError(CraftError):
+    """The build plan filtered out all possible builds."""
+
+    def __init__(self) -> None:
+        message = "No build matches the current platform."
+        resolution = 'Check the "--platform" and "--build-for" parameters.'
+
+        super().__init__(message=message, resolution=resolution)
+
+
+class MultipleBuildsError(CraftError):
+    """The build plan contains multiple possible builds."""
+
+    def __init__(self) -> None:
+        message = "Multiple builds match the current platform."
+        resolution = 'Check the "--platform" and "--build-for" parameters.'
```

Comment:
`--build-for` as a CLI option is Snapcraft specific

Reply:
I dropped the "--build-for" parameter from the resolution, to keep the error more platform-centric